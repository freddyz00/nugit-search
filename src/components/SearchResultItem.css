.search-result-item {
    padding: 2rem 0;
    border-top: 1px solid lightgray;
}

.search-result-item__title {
    display: flex;
    align-items: flex-end;
    text-decoration: none;
    font-weight: normal;
    color: #4040c4;
    margin-top: 0;
}

.search-result-item a {
    color: #4040c4;
    text-decoration: inherit;
}

.search-result-item a:hover {
    text-decoration: underline;
}

.search-result-item__name {
    font-weight: bold;
}

.search-result-item__language {
    font-size: 0.75rem;
    padding: 0.25rem 0.75rem;
    border-radius: 99px;
    color: black;
    margin-left: 1.5rem;
    border: 1px solid lightgray;
}

.search-result-item__topics {
    display: flex;
    flex-wrap: wrap;

}

.search-result-item__topics a {
    font-size: 0.8rem;
    padding: 0.25rem 0.75rem;
    color: #0041f3;
    background-color: #c8e3ff;
    border-radius: 99px;
    margin-right: 0.75rem;
    margin-bottom: 0.5rem;
    text-decoration: none;
}

.search-result-item__topics a:hover {
    text-decoration: none;
    background-color: #007bff;
    color: white;
}