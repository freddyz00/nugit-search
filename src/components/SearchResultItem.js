import React from "react";
import "./SearchResultItem.css";

function SearchResultItem({
  name,
  owner,
  description,
  language,
  htmlUrl,
  topics,
}) {
  return (
    <div className="search-result-item">
      {/* full name of the repository */}
      <h2 className="search-result-item__title">
        <a href={htmlUrl}>
          <span className="search-result-item__owner">{`${owner} / `}</span>
          <span className="search-result-item__name">{name}</span>
        </a>

        {/* language of the repository */}
        {language && (
          <span className="search-result-item__language">{language}</span>
        )}
      </h2>

      {/* description of the repository */}
      <p>{description}</p>

      {/* topics of the repository */}
      <div className="search-result-item__topics">
        {topics.map((topic, index) => (
          <a href={`http://www.github.com/topics/${topic}`} key={index}>
            {topic}
          </a>
        ))}
      </div>
    </div>
  );
}

export default SearchResultItem;
