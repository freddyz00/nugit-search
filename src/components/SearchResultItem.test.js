import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchResultItem from "./SearchResultItem";

describe("Search Result Item", () => {
  it("renders correct information", () => {
    const tree = renderer
      .create(
        <SearchResultItem
          name="name"
          owner="owner"
          description="This is a description"
          language="JavaScript"
          htmlUrl="https://www.github.com"
          topics={["react", "javascript"]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
