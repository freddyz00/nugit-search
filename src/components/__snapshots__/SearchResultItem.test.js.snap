// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result Item renders correct information 1`] = `
<div
  className="search-result-item"
>
  <h2
    className="search-result-item__title"
  >
    <a
      href="https://www.github.com"
    >
      <span
        className="search-result-item__owner"
      >
        owner / 
      </span>
      <span
        className="search-result-item__name"
      >
        name
      </span>
    </a>
    <span
      className="search-result-item__language"
    >
      JavaScript
    </span>
  </h2>
  <p>
    This is a description
  </p>
  <div
    className="search-result-item__topics"
  >
    <a
      href="http://www.github.com/topics/react"
    >
      react
    </a>
    <a
      href="http://www.github.com/topics/javascript"
    >
      javascript
    </a>
  </div>
</div>
`;
